checkov/terraform/modules/module_objects.py

```diff
@@ -4,41 +4,30 @@
 
 
 class TFModule:
-    __slots__ = ("path", "name", "foreach_idx")
+    __slots__ = ("path", "name", "foreach_idx", "nested_tf_module")
 
-    def __init__(self, path: str, name: str, foreach_idx: Optional[int | str] = None) -> None:
+    def __init__(self, path: str, name: str, nested_tf_module: Optional[TFModule] = None, foreach_idx: Optional[int | str] = None) -> None:
         self.path = path
         self.name = name
         self.foreach_idx = foreach_idx
+        self.nested_tf_module = nested_tf_module
 
     def __eq__(self, other: Any) -> bool:
         if not isinstance(other, TFModule):
             return False
-        return self.path == other.path and self.name == other.name and self.foreach_idx == other.foreach_idx
+        return self.path == other.path and self.name == other.name and self.nested_tf_module == other.nested_tf_module and self.foreach_idx == other.foreach_idx
 
     def __repr__(self) -> str:
-        return f'path:{self.path}, name:{self.name}, foreach_idx:{self.foreach_idx}'
+        return f'path:{self.path}, name:{self.name}, nested_tf_module:{self.nested_tf_module}, foreach_idx:{self.foreach_idx}'
 
-
-class TFSourceModules:
-    __slots__ = ("tf_source_modules", )
-
```

Comment:
Why do we remove the class?

Reply:
We will use the TFModule object with TFModule member for the nested modules